torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -266,6 +267,67 @@ class SchedulerTopologyChecker {
     }
     return false;
   }
+
+  // Checks if there's any unsupported operations post reduction. If outer
+  // reduction we can fuse some pointwise ops if they don't require
+  // broadcasting. For inner reductions we cannot fuse any binary like
+  // operations. Included other operations we're not even fusing at the moment
+  // like transpose and shift.
+  static bool supportedPostReductionFusion(
+      Fusion* fusion,
+      std::vector<TensorView*> reduction_tvs) {
+    TORCH_INTERNAL_ASSERT(reduction_tvs.size());
+    bool fastest_dim_reduction = true;
+    auto red_root_dom = reduction_tvs[0]->getRootDomain();
+    for (size_t i = red_root_dom.size(); i > 0; i--) {
+      if (red_root_dom[i - 1]->isBroadcast() ||
+          red_root_dom[i - 1]->isTrivialReduction()) {
+        continue;
+      } else if (red_root_dom[i - 1]->isReduction()) {
+        fastest_dim_reduction = true;
+        break;
+      } else {
+        fastest_dim_reduction = false;
+        break;
+      }
+    }
+
+    // If reduction is on fastest dim, don't fuse any non Unary or Broadcast
+    // operations post reduction.
```

Comment:
Nit, slightly different, applies if we use another input in these operations (not an input to the reductions).